contributions/catalog/controllers/contribute.py

```diff
@@ -74,7 +76,64 @@ def result():
 @bp.route('details/<contribution_id>', methods=['GET'])
 def contribution_details(contribution_id):
     the_json_res = get_contribution(contribution_id)
-    return render_template("contribute/contribution_details.html", post=the_json_res, user=session["name"])
+
+    # check to see if the logged in user is an editable user for creating edit button
+    is_editable = False
+    username = session["username"]
+    headers = requestutil.get_header_using_session(session)
+    is_editable = adminutil.check_if_reviewer(username, headers)
+
+    return render_template("contribute/contribution_details.html", is_editable=is_editable, post=the_json_res, user=session["name"])
+
+@bp.route('create/<contribution_id>/edit', methods=['GET', 'POST'])
+@login_required
+def contribution_edit(contribution_id):
+    if request.method == 'POST':
+        is_put = False
+        contribution_id = None
+        result = request.form.to_dict(flat=False)
+
+        # check if it is PUT
+        try:
+            contribution_id = result["contribution_id"][0]
+            is_put = True
+        except:
+            s = "There is a error in edit. The method is not an edit."
+            return render_template('contribute/error.html', error_msg=s)
+
+        if is_put:
+            contribution = to_contribution(result)
+            contribution = jsonutil.add_contribution_admins(contribution, is_edit=True)
+            # remove id from json_data
+            del contribution["id"]
+            json_contribution = json.dumps(contribution, indent=4)
+            response, s = put_contribution(json_contribution, contribution_id)
+
+            if response:
+                if response:
+                    if "name" in session:
+                        return render_template('contribute/submitted.html', user=session["name"],  token=session['oauth_token']['access_token'])
+                    else:
+                        return render_template('contribute/submitted.html')
+                elif not response:
+                    if "name" in session:
+                        return render_template('contribute/error.html', user=session["name"],  token=session['oauth_token']['access_token'], error_msg=s)
+                    else:
+                        return render_template('contribute/error.html', error_msg=s)
+    else:
+        the_json_res = get_contribution(contribution_id)
+        # check if the user is editable then set the is_editable
+        is_editable = False
+        username = session["username"]
+        headers = requestutil.get_header_using_session(session)
+        is_editable = adminutil.check_if_reviewer(username, headers)
+
+        if is_editable:
+            return render_template('contribute/contribute.html', is_editable=is_editable, user=session["name"], token=session['oauth_token']['access_token'], post=the_json_res)
```

Comment:
I think it will be better to redirect the user to the contribution detailed view page, after submitting the edit page. Thanks.

Reply:
This might mess up the other PR as well. I will create a issue for that.